### ISSUE TITLE
Create order api not saving customer_id right

### ISSUE THREAD

**Issue by PidgeyBE** [woocommerce/woocommerce#21741]
Hi all!

I'm investigated if I can use woocommerce API as backend for a webstore I'm trying to make (integrated into other software).
I'm using the latest wordpress with latest Woocommerce + Dokan multivendor plugin installed.
I'm using JWT Authentication plugin to get access to the API.

When I add an order via the POST `wp-json/wc/v3/orders` endpoint, the order is not created with the right "customer_id". Also trying the edit it via `wp-json/wc/v3/orders/id` doesn't work. "

Changing the id via the WP admin panel UI works.

## Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate
- [X] The issue still exists against the latest `master` branch of WooCommerce on Github (this is **not** the same version as on WordPress.org!)
- [X] I have attempted to find the simplest possible steps to reproduce the issue
- [ ] I have included a failing test as a pull request (Optional)

## Steps to reproduce the issue

1. POST to http://localhost:8000/wp-json/wc/v3/orders/
`{
    "payment_method": "bacs",
    "payment_method_title": "Direct Bank Transfer",
    "set_paid": false,
    "customer_id": 3,
    "billing": {
        "first_name": "John2",
        "last_name": "Doe",
        "address_1": "969 Market",
        "address_2": "",
        "city": "San Francisco",
        "state": "CA",
        "postcode": "94103",
        "country": "US",
        "email": "john.doe@example.com",
        "phone": "(555) 555-5555"
    },
    "shipping": {
        "first_name": "Joh2n",
        "last_name": "Doe",
        "address_1": "969 Market",
        "address_2": "",
        "city": "San Francisco",
        "state": "CA",
        "postcode": "94103",
        "country": "US"
    },
    "line_items": [
        {
            "product_id": 21,
            "quantity": 2
        }
    ]
}`
2. -> reply answers with right "customer_id: 3".
3. However, when looking inside the woocommerce UI or when doing GET `http://localhost:8000/wp-json/wc/v3/orders/31`, I see "customer_id": 1"!

## Expected/actual behavior
=> "customer_id" key is not setable via the `wp-json/wc/v3/orders` endpoint. It seems the endpoint takes the id of the account making the API call. (in my case admin account, with id 1).

Expected behaviour is that I could make orders for other users by setting the "customer_id".

## Isolating the problem

- [ ] This bug happens with only WooCommerce plugin active
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/)
- [X] I can reproduce this bug consistently using the steps above

## WordPress Environment

<details>
```
`
### WordPress Environment ###

Home URL: http://localhost:8000
Site URL: http://localhost:8000
WC Version: 3.5.0
Log Directory Writable: ✔
WP Version: 4.9.8
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: Apache/2.4.25 (Debian)
PHP Version: 7.2.11
PHP Post Max Size: 8 MB
PHP Time Limit: 30
PHP Max Input Vars: 1000
cURL Version: 7.52.1
OpenSSL/1.0.2l

SUHOSIN Installed: –
MySQL Version: 5.7.23
Max Upload Size: 2 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ❌ Your server does not have the SoapClient class enabled - some gateway plugins which use SOAP may not work as expected.
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.5.0
WC Database Prefix: wp_
MaxMind GeoIP Database: ❌ The MaxMind GeoIP Database does not exist - Geolocation will not function. You can download and install it manually from https://dev.maxmind.com/geoip/geoip2/geolite2/ to the path: . Scroll down to "Downloads" and download the "MaxMind DB binary
gzipped" file next to "GeoLite2 Country". Please remember to uncompress GeoLite2-Country_xxxxxxxx.tar.gz and upload the GeoLite2-Country.mmdb file only.

Total Database Size: 3.12MB
Database Data Size: 2.21MB
Database Index Size: 0.91MB
wp_woocommerce_sessions: Data: 0.05MB + Index: 0.02MB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.05MB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB
wp_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB
wp_comments: Data: 0.02MB + Index: 0.09MB
wp_dokan_announcement: Data: 0.02MB + Index: 0.00MB
wp_dokan_orders: Data: 0.02MB + Index: 0.03MB
wp_dokan_refund: Data: 0.02MB + Index: 0.00MB
wp_dokan_vendor_balance: Data: 0.02MB + Index: 0.00MB
wp_dokan_withdraw: Data: 0.02MB + Index: 0.00MB
wp_failed_jobs: Data: 0.02MB + Index: 0.00MB
wp_links: Data: 0.02MB + Index: 0.02MB
wp_mailchimp_carts: Data: 0.02MB + Index: 0.00MB
wp_options: Data: 1.44MB + Index: 0.05MB
wp_postmeta: Data: 0.06MB + Index: 0.03MB
wp_posts: Data: 0.02MB + Index: 0.06MB
wp_queue: Data: 0.02MB + Index: 0.00MB
wp_termmeta: Data: 0.02MB + Index: 0.03MB
wp_terms: Data: 0.02MB + Index: 0.03MB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB
wp_usermeta: Data: 0.02MB + Index: 0.03MB
wp_users: Data: 0.02MB + Index: 0.05MB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB

### Post Type Counts ###

attachment: 6
customize_changeset: 1
page: 11
post: 2
product: 7
shop_order: 1

### Security ###

Secure connection (HTTPS): ❌
					Your store is not using HTTPS. Learn more about HTTPS and SSL Certificates.
Hide errors from visitors: ✔

### Active Plugins (4) ###

Dokan: by weDevs
LLC – 2.9.2

Jetpack by WordPress.com: by Automattic – 6.6.1
JWT Authentication for WP-API: by Enrique Chavez – 1.2.4
Mailchimp for WooCommerce: by Mailchimp – 2.1.11 – Not tested with the active version of WooCommerce

### Settings ###

API Enabled: ✔
Force SSL: –
Currency: EUR (€)
Currency Position: left
Thousand Separator: 
Decimal Separator: ,
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)


### WC Pages ###

Shop base: #5 - /shop/
Cart: #6 - /cart/
Checkout: #7 - /checkout/
My account: #8 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Storefront
Version: 2.3.5
Author URL: https://woocommerce.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –
`
```
</details>

**Comment by rodrigoprimo** [comment]
This is likely caused by a bug that was introduced in 3.5.0. We are reverting the changes that caused this problem in 3.5.1 that will be released today. Closing this issue for now. Please leave a comment and I will reopen if 3.5.1 doesn't fix the issue. Thanks.

**Comment by rodrigoprimo** [comment]
See #21740 for more information.

**Comment by PidgeyBE** [comment]
Indeed fixed in 3.5.1! thanks!
